base_decorator.rb

```diff
@@ -0,0 +1,15 @@
+require './nameable'
+
+class BaseDecoraroe < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] (OPTIONAL) Not really optional, kindly add the `attr_accessor :nameable` to this class. 👍